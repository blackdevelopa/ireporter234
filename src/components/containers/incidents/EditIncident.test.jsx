import React from 'react';
import { mount } from 'enzyme';
import thunk from 'redux-thunk';
import { MemoryRouter as Router } from 'react-router-dom';
import configMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import ConnectedEditIncident, { EditOneIncident } from './EditIncident';

const props = {
  editSingleIncident: jest.fn(),
};

const initialState = {
  isLoading: false,
  error: false,
  incident: [],
  newIncident: false,
  singleIncident: {},
  editIncident: {},
};

jest.mock('../../../redux/actions/incident/incident');
jest.mock('../../../store/');
const mockStore = configMockStore([thunk]);
const store = mockStore(initialState);

const MyContext = React.createContext();

const wrapper = mount(
  <Provider store={store} context={MyContext}>
    <Router>
      <ConnectedEditIncident context={MyContext} />
    </Router>
  </Provider>
);

describe('<EditIncident />', () => {
  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call editIncidence for location handleOnChange', () => {
    const editIncident = mount(<EditOneIncident {...props} />);
    const spy = jest.spyOn(editIncident.instance(), 'handleInputChange');
    const event = { target: { name: 'location', value: 'location' } };
    editIncident.instance().forceUpdate();
    editIncident.find('input[name="location"]').simulate('change', event);
    expect(spy).toBeCalled();
    expect(editIncident.instance().state.location).toBe('location');
  });

  it('should test that editIncident form is submitted', () => {
    const editIncident = mount(<EditOneIncident {...props} />);
    const form = editIncident.find('Form');
    const spy = jest.spyOn(editIncident.instance(), 'handleSubmit');
    editIncident.instance().forceUpdate();
    form.simulate('submit');
    expect(spy).toBeCalled();
  });
});
