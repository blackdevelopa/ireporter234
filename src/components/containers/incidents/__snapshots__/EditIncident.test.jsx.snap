// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditIncident /> should render without crashing 1`] = `
<Provider
  context={
    Object {
      "$$typeof": Symbol(react.context),
      "Consumer": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": [Circular],
      },
      "Provider": Object {
        "$$typeof": Symbol(react.provider),
        "_context": [Circular],
      },
      "_calculateChangedBits": null,
      "_currentRenderer": Object {},
      "_currentRenderer2": null,
      "_currentValue": undefined,
      "_currentValue2": undefined,
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "hwjq02",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "hwjq02",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(withRouter(EditOneIncident))
        context={
          Object {
            "$$typeof": Symbol(react.context),
            "Consumer": Object {
              "$$typeof": Symbol(react.context),
              "_calculateChangedBits": null,
              "_context": [Circular],
            },
            "Provider": Object {
              "$$typeof": Symbol(react.provider),
              "_context": [Circular],
            },
            "_calculateChangedBits": null,
            "_currentRenderer": Object {},
            "_currentRenderer2": null,
            "_currentValue": undefined,
            "_currentValue2": undefined,
          }
        }
      >
        <withRouter(EditOneIncident)
          context={
            Object {
              "$$typeof": Symbol(react.context),
              "Consumer": Object {
                "$$typeof": Symbol(react.context),
                "_calculateChangedBits": null,
                "_context": [Circular],
              },
              "Provider": Object {
                "$$typeof": Symbol(react.provider),
                "_context": [Circular],
              },
              "_calculateChangedBits": null,
              "_currentRenderer": Object {},
              "_currentRenderer2": null,
              "_currentValue": undefined,
              "_currentValue2": undefined,
            }
          }
          editSingleIncident={[Function]}
        >
          <Route>
            <EditOneIncident
              context={
                Object {
                  "$$typeof": Symbol(react.context),
                  "Consumer": Object {
                    "$$typeof": Symbol(react.context),
                    "_calculateChangedBits": null,
                    "_context": [Circular],
                  },
                  "Provider": Object {
                    "$$typeof": Symbol(react.provider),
                    "_context": [Circular],
                  },
                  "_calculateChangedBits": null,
                  "_currentRenderer": Object {},
                  "_currentRenderer2": null,
                  "_currentValue": undefined,
                  "_currentValue2": undefined,
                }
              }
              editSingleIncident={[Function]}
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "hwjq02",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "hwjq02",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "key": "hwjq02",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <div>
                <Navbar
                  access="true"
                  firstbtn="Redflag"
                  secondbtn="Intervention"
                >
                  <div>
                    <div>
                      <a
                        href="/"
                      >
                        iReporter
                      </a>
                      <Button
                        as="button"
                        color="blue"
                        compact={true}
                        size="medium"
                      >
                        <Ref
                          innerRef={
                            Object {
                              "current": <button
                                class="ui blue medium compact button"
                              >
                                Redflag
                              </button>,
                            }
                          }
                        >
                          <RefFindNode
                            innerRef={
                              Object {
                                "current": <button
                                  class="ui blue medium compact button"
                                >
                                  Redflag
                                </button>,
                              }
                            }
                          >
                            <button
                              className="ui blue medium compact button"
                              onClick={[Function]}
                            >
                              Redflag
                            </button>
                          </RefFindNode>
                        </Ref>
                      </Button>
                      <Button
                        as="button"
                        color="green"
                        compact={true}
                        size="medium"
                      >
                        <Ref
                          innerRef={
                            Object {
                              "current": <button
                                class="ui green medium compact button"
                              >
                                Intervention
                              </button>,
                            }
                          }
                        >
                          <RefFindNode
                            innerRef={
                              Object {
                                "current": <button
                                  class="ui green medium compact button"
                                >
                                  Intervention
                                </button>,
                              }
                            }
                          >
                            <button
                              className="ui green medium compact button"
                              onClick={[Function]}
                            >
                              Intervention
                            </button>
                          </RefFindNode>
                        </Ref>
                      </Button>
                    </div>
                  </div>
                </Navbar>
                <switchNav>
                  <div
                    props={Object {}}
                  />
                </switchNav>
                <Container>
                  <div
                    className="ui container"
                  >
                    <div>
                      <strong>
                        Update Incident
                      </strong>
                    </div>
                    <Form
                      as="form"
                      onSubmit={[Function]}
                    >
                      <form
                        className="ui form"
                        onSubmit={[Function]}
                      >
                        <FormGroup
                          widths="equal"
                        >
                          <div
                            className="equal width fields"
                          >
                            <FormField
                              control={[Function]}
                              label="Your Location"
                              name="location"
                              onChange={[Function]}
                              placeholder="lagos..."
                            >
                              <div
                                className="field"
                              >
                                <label>
                                  Your Location
                                </label>
                                <Input
                                  name="location"
                                  onChange={[Function]}
                                  placeholder="lagos..."
                                  type="text"
                                >
                                  <div
                                    className="ui input"
                                  >
                                    <input
                                      name="location"
                                      onChange={[Function]}
                                      placeholder="lagos..."
                                      type="text"
                                    />
                                  </div>
                                </Input>
                              </div>
                            </FormField>
                          </div>
                        </FormGroup>
                        <FormField
                          control={[Function]}
                          label="Comment"
                          name="comment"
                          onChange={[Function]}
                          placeholder="Share what you can see..."
                        >
                          <div
                            className="field"
                          >
                            <label>
                              Comment
                            </label>
                            <TextArea
                              as="textarea"
                              name="comment"
                              onChange={[Function]}
                              placeholder="Share what you can see..."
                              rows={3}
                            >
                              <Ref
                                innerRef={
                                  Object {
                                    "current": <textarea
                                      name="comment"
                                      placeholder="Share what you can see..."
                                      rows="3"
                                    />,
                                  }
                                }
                              >
                                <RefFindNode
                                  innerRef={
                                    Object {
                                      "current": <textarea
                                        name="comment"
                                        placeholder="Share what you can see..."
                                        rows="3"
                                      />,
                                    }
                                  }
                                >
                                  <textarea
                                    name="comment"
                                    onChange={[Function]}
                                    onInput={[Function]}
                                    placeholder="Share what you can see..."
                                    rows={3}
                                  />
                                </RefFindNode>
                              </Ref>
                            </TextArea>
                          </div>
                        </FormField>
                        <FormField
                          control={[Function]}
                          fluid={true}
                        >
                          <div
                            className="field"
                          >
                            <Button
                              as="button"
                              fluid={true}
                            >
                              <Ref
                                innerRef={
                                  Object {
                                    "current": <button
                                      class="ui fluid button"
                                    >
                                      Submit
                                    </button>,
                                  }
                                }
                              >
                                <RefFindNode
                                  innerRef={
                                    Object {
                                      "current": <button
                                        class="ui fluid button"
                                      >
                                        Submit
                                      </button>,
                                    }
                                  }
                                >
                                  <button
                                    className="ui fluid button"
                                    onClick={[Function]}
                                  >
                                    Submit
                                  </button>
                                </RefFindNode>
                              </Ref>
                            </Button>
                          </div>
                        </FormField>
                      </form>
                    </Form>
                  </div>
                </Container>
              </div>
            </EditOneIncident>
          </Route>
        </withRouter(EditOneIncident)>
      </Connect(withRouter(EditOneIncident))>
    </Router>
  </MemoryRouter>
</Provider>
`;
